aws-xray-auto-instrumentation-agent-runtime/src/main/java/com/amazonaws/xray/agent/config/XRaySDKConfiguration.java

```diff
@@ -1,104 +1,322 @@
+        return agentConfiguration;
+    }
+
+    // For testing
+    void setAgentConfiguration(AgentConfiguration agentConfiguration) {
+        this.agentConfiguration = agentConfiguration;
+    }
+
+    public boolean isEnabled() {
+        return agentConfiguration.isTracingEnabled();
+    }
+
+    private XRaySDKConfiguration() {
+    }
 
     /**
      * @return XRaySDKConfiguration - The global instance of this agent recorder configuration.
      */
     public static XRaySDKConfiguration getInstance() {
+        if (instance == null) {
+            instance = new XRaySDKConfiguration();
+        }
         return instance;
     }
 
     /**
-     * Initialize the Agent Recorder with the set attributes.
-     * @throws MalformedURLException Thrown if no sampling strategy was passed in and the sampling rule file path is invalid.
+     * Parses the given agent configuration file and stores its properties. If file is missing or incorrectly formatted,
+     * the agent is configured with the default settings.
+     * @param configFile - Location of configuration file
      */
-    public void init() throws MalformedURLException {
+    public void init(URL configFile) {
+        try {
+            log.info("Reading X-Ray Agent config file at: " + configFile.getPath());
+            this.agentConfiguration = parseConfig(configFile);
+        } catch (IOException e) {
+            throw new InvalidAgentConfigException("Failed to read X-Ray Agent configuration file " + configFile.getPath(), e);
+        }
+
+        if (log.isDebugEnabled()) {
+            log.debug("Starting the X-Ray Agent with the following properties:\n" + agentConfiguration.toString());
+        }
+        init();
+    }
+
+    /**
+     * Initialize the X-Ray SDK's Recorder used by the agent.
+     */
+    public void init() {
+        init(AWSXRayRecorderBuilder.standard());
+    }
+
+    // For testing only
+    void init(AWSXRayRecorderBuilder builder) {
         log.info("Initializing the X-Ray Agent Recorder");
-        AWSXRayRecorderBuilder builder = AWSXRayRecorderBuilder.standard();
-        builder.withThrowableSerializationStrategy(new DefaultThrowableSerializationStrategy(maxStackTraceLength));
-        if (segmentContextResolverChain != null) {
-            builder.withSegmentContextResolverChain(segmentContextResolverChain);
+
+        // Reset to defaults
+        if (agentConfiguration == null) {
+            agentConfiguration = new AgentConfiguration();
         }
-        if (contextMissingStrategy != null) {
-            builder.withContextMissingStrategy(contextMissingStrategy);
+
+        this.awsServiceHandlerManifest = null;
+        this.awsSDKVersion = 0;
+
+        // X-Ray Enabled
+        String envString = System.getenv(ENABLED_ENVIRONMENT_VARIABLE_KEY);
+        String sysString = System.getProperty(ENABLED_SYSTEM_PROPERTY_KEY);
+        if (envString != null && envString.toLowerCase().equals("false") ||
```

Comment:
Believe these can be simplified to `"false".equalsIgnoreCase(envString)`

Reply:
Good trick!